cpp/src/utilities/error_utils.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef ERRORUTILS_HPP
+#define ERRORUTILS_HPP
+
+#include <cuda.h>
+#include <cuda_runtime_api.h>
+#include <iostream>
+#include <stdexcept>
+
+#define RMM_TRY(x) \
+  if ((x) != RMM_SUCCESS) return GDF_MEMORYMANAGER_ERROR;
+
```

Comment:
Should we label these as deprecated? And point developers to the macros they should be using instead. For example GDF_REQUIRE is to be  replaced by CUDF_EXPECTS right? I think that would help

Reply:
I don't think you can change GDF_REQUIRE to CUDF_EXPECTS without changing the calling function's signature to not return a gdf_error, right?